README.md

```diff
@@ -207,31 +162,131 @@ Users want to determine several types of the formatting width as following
+
+### Display fractional values
+
+Sometimes it is desirable to display the smallest sub-second unit not by itself but as a fraction of the immediately larger unit.
 
-Adding the locale in `string` format as a first argument.
+#### Design
 
-# API design
+We allow users to specify a `fractionalDigits` option that will display the smallest sub-second unit with display set to `"auto"` as a fraction of the previous unit if it is non-zero. The number of digits used will be the value passed to this option.
+
+#### Example
+
+```javascript
+new Intl.DurationFormat('en', { fractionalDigits: 2 }).format('PT12.3456S'); // => 12.34 sec
+new Intl.DurationFormat('en', { milliseconds: 'narrow', fractionalDigits: 2 }).format('PT12.3456S'); // => 12 sec 345.60 ms
+```
+
+## API design
 
 ``` javascript
-    const formatter =
-        new Intl.DurationFormat('en-US', {
-            fields: [
-                'hour',
-                'minute',
-                'second'
-            ],
-            style: 'short',
-            hideZeroValues: 'none',
-        });
-
-    const duration =
-        Temporal.Duration.from({
-            hours: 2,
-            minutes: 46,
-            seconds: 40
-        });
-
-    console.log(formatter.format(duration));
-    // output: 2 hr 46 min 40 sec
+    new Intl.DurationFormat('en').format(Temporal.Duration.from('PT2H46M40S')); // => 2 hr 46 min 40 sec
+    new Intl.DurationFormat('en', {
+      hours: 'numeric',
+      seconds: 'numeric',
+    }).format(Temporal.Duration.from('PT2H40S')); // => 2:00:40
+```
+
+### Constructor
+
+#### Syntax
+
+```javascript
+new Intl.DurationFormat(locales, options)
 ```
+
+#### Parameters
+
+* `locales: Array<string> | string`: A locale string or a list of locale strings in decreasing order of preference.
+* `options?: object`: An object for configuring the behavior of the instance. It may have some or all of the following properties:
+  * `localeMatcher: "best-fit" | "lookup"`: A string denoting which locale matching algorithm to use. Defaults to `"best fit"`.
+  * `numberingSystem: string`: A string containing the name of the numbering system to be used for number formatting.
+  * `style: "long" | "short" | "narrow"`: The base style to be used for formatting. This can be overriden per-unit by setting the more granular options. Defaults to `"short"`.
+  * `years: "long" | "short" | "narrow"`: The style to be used for formatting years.
+  * `yearsDisplay: "always" | "auto"`: Whether to always display years, or only if nonzero.
+  * `months: "long" | "short" | "narrow"`: The style to be used for formatting months.
+  * `monthsDisplay: "always" | "auto"`: Whether to always display months, or only if nonzero.
+  * `weeks: "long" | "short" | "narrow"`: The style to be used for formatting weeks.
+  * `weeksDisplay: "always" | "auto"`: Whether to always display weeks, or only if nonzero.
+  * `days: "long" | "short" | "narrow"`: The style to be used for formatting days.
```

Comment:
per my comment elsewhere, IMHO we should offer the `'numeric'` style for `'days'` too, to support NASA-style countdown timers that include days.

Reply:
I suppose we could merge the current text as per the current consensus and sneak this quick change once we have a decision for that issue?

Reply:
Yep, that works. Or we could add it now and remove it later if it's not desired. Regardless, I opened an issue to track: https://github.com/tc39/proposal-intl-duration-format/issues/67